[Bug]: `dateutil` time parser in OpenEphysLegacy

### What happened?

Issue raised in https://github.com/catalystneuro/tye-lab-to-nwb/issues/28, dateutil parser cannot parse '29-Apr-2020 16554'
Switching to `datetime.strptime` with the expected format "%d-%b-%Y %H%M%S" will parse this as `2020-04-29 16:55:04`



### Steps to Reproduce

```python
import dateutil
dateutil.parser.parse('29-Apr-2020 16554')
```


### Traceback

```python
../../../anaconda3/envs/neuroconv311/lib/python3.11/site-packages/dateutil/parser/_parser.py:1368: in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <dateutil.parser._parser.parser object at 0x13dbed090>
timestr = '29-Apr-2020 16554', default = datetime.datetime(2023, 9, 22, 0, 0)
ignoretz = False, tzinfos = None, kwargs = {}, res = None, skipped_tokens = None

    def parse(self, timestr, default=None,
              ignoretz=False, tzinfos=None, **kwargs):
        """
        Parse the date/time string into a :class:`datetime.datetime` object.
    
        :param timestr:
            Any date/time string using the supported formats.
    
        :param default:
            The default datetime object, if this is a datetime object and not
            ``None``, elements specified in ``timestr`` replace elements in the
            default object.
    
        :param ignoretz:
            If set ``True``, time zones in parsed strings are ignored and a
            naive :class:`datetime.datetime` object is returned.
    
        :param tzinfos:
            Additional time zone names / aliases which may be present in the
            string. This argument maps time zone names (and optionally offsets
            from those time zones) to time zones. This parameter can be a
            dictionary with timezone aliases mapping time zone names to time
            zones or a function taking two parameters (``tzname`` and
            ``tzoffset``) and returning a time zone.
    
            The timezones to which the names are mapped can be an integer
            offset from UTC in seconds or a :class:`tzinfo` object.
    
            .. doctest::
               :options: +NORMALIZE_WHITESPACE
    
                >>> from dateutil.parser import parse
                >>> from dateutil.tz import gettz
                >>> tzinfos = {"BRST": -7200, "CST": gettz("America/Chicago")}
                >>> parse("2012-01-19 17:21:00 BRST", tzinfos=tzinfos)
                datetime.datetime(2012, 1, 19, 17, 21, tzinfo=tzoffset(u'BRST', -7200))
                >>> parse("2012-01-19 17:21:00 CST", tzinfos=tzinfos)
                datetime.datetime(2012, 1, 19, 17, 21,
                                  tzinfo=tzfile('/usr/share/zoneinfo/America/Chicago'))
    
            This parameter is ignored if ``ignoretz`` is set.
    
        :param \\*\\*kwargs:
            Keyword arguments as passed to ``_parse()``.
    
        :return:
            Returns a :class:`datetime.datetime` object or, if the
            ``fuzzy_with_tokens`` option is ``True``, returns a tuple, the
            first element being a :class:`datetime.datetime` object, the second
            a tuple containing the fuzzy tokens.
    
        :raises ParserError:
            Raised for invalid or unknown string format, if the provided
            :class:`tzinfo` is not in a valid format, or if an invalid date
            would be created.
    
        :raises TypeError:
            Raised for non-string or character stream input.
    
        :raises OverflowError:
            Raised if the parsed date exceeds the largest valid C integer on
            your system.
        """
    
        if default is None:
            default = datetime.datetime.now().replace(hour=0, minute=0,
                                                      second=0, microsecond=0)
    
        res, skipped_tokens = self._parse(timestr, **kwargs)
    
        if res is None:
>           raise ParserError("Unknown string format: %s", timestr)
E           dateutil.parser._parser.ParserError: Unknown string format: 29-Apr-2020 16554
```


### Operating System

macOS

### Python Executable

Python

### Python Version

3.9

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,8 @@
 # (Upcoming)
 
+### Fixes
 
+* Changed the date parsing in `OpenEphysLegacyRecordingInterface` to `datetime.strptime` with the expected date format explicitly set to `"%d-%b-%Y %H%M%S"`.
 
 # v0.4.4
 

--- a/src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py
+++ b/src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py
@@ -70,14 +70,15 @@ class OpenEphysLegacyRecordingInterface(BaseRecordingExtractorInterface):
             if segment_annotations:
                 expected_date_format = "%d-%b-%Y %H%M%S"
                 date_created = segment_annotations[0]["date_created"]
-                try:
-                    date_created = date_created.strip("'")
-                    session_start_time = datetime.strptime(date_created, expected_date_format)
-                    metadata["NWBFile"].update(session_start_time=session_start_time)
-                except ValueError:
+                date_created = date_created.strip("'")
+                extracted_date, extracted_timestamp = date_created.split(" ")
+                if len(extracted_timestamp) != len("%H%M%S"):
                     warn(
-                        "The 'session_start_time' could not be extracted from the header."
-                        f"Received '{date_created}' which does not match the expected format ({expected_date_format})."
+                        f"The timestamp for starting time from openephys metadata is ambiguous ('{extracted_timestamp}')! Only the date will be auto-populated in metadata. Please update the timestamp manually to record this value with the highest known temporal resolution."
                     )
+                    session_start_time = datetime.strptime(extracted_date, "%d-%b-%Y")
+                else:
+                    session_start_time = datetime.strptime(date_created, expected_date_format)
+                metadata["NWBFile"].update(session_start_time=session_start_time)
 
         return metadata

--- a/src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py
+++ b/src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py
@@ -1,6 +1,6 @@
+from datetime import datetime
 from typing import List, Optional
-
-from dateutil.parser import parse
+from warnings import warn
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ....utils import FolderPathType
@@ -68,7 +68,14 @@ class OpenEphysLegacyRecordingInterface(BaseRecordingExtractorInterface):
         if block_annotations:
             segment_annotations = block_annotations[0].get("segments", [])
             if segment_annotations:
+                expected_date_format = "%d-%b-%Y %H%M%S"
                 date_created = segment_annotations[0]["date_created"]
-                session_start_time = parse(date_created)
-                metadata["NWBFile"].update(session_start_time=session_start_time)
+                try:
+                    date_created = date_created.strip("'")
+                    session_start_time = datetime.strptime(date_created, expected_date_format)
+                    metadata["NWBFile"].update(session_start_time=session_start_time)
+                except ValueError:
+                    warn("The 'session_start_time' could not be extracted from the header."
+                         f"Received '{date_created}' which does not match the expected format ({expected_date_format}).")
+
         return metadata

--- a/tests/test_on_data/test_gin_ecephys/test_openephyslegacy.py
+++ b/tests/test_on_data/test_gin_ecephys/test_openephyslegacy.py
@@ -41,6 +41,11 @@ class TestOpenEphysLegacyConversions(TestCase):
         segments = blocks[0].get("segments", [])
         # Override date_created to test date is parsed
         segments[0].update(date_created="29-Apr-2020 16554")
-        metadata = self.interface.get_metadata()
+        with self.assertWarnsWith(
+            UserWarning,
+            exc_msg="The timestamp for starting time from openephys metadata is ambiguous ('16554')! Only the date will be auto-populated in metadata. Please update the timestamp manually to record this value with the highest known temporal resolution.",
+        ):
+            metadata = self.interface.get_metadata()
+
         assert "session_start_time" in metadata["NWBFile"]
-        self.assertEqual(metadata["NWBFile"]["session_start_time"], datetime(2020, 4, 29, 16, 55, 4))
+        self.assertEqual(metadata["NWBFile"]["session_start_time"], datetime(2020, 4, 29, 0, 0))

--- a/tests/test_on_data/test_gin_ecephys/test_openephyslegacy.py
+++ b/tests/test_on_data/test_gin_ecephys/test_openephyslegacy.py
@@ -1,3 +1,5 @@
+from datetime import datetime
+
 from hdmf.testing import TestCase
 
 from neuroconv.datainterfaces.ecephys.openephys.openephyslegacydatainterface import (
@@ -30,3 +32,16 @@ class TestOpenEphysLegacyConversions(TestCase):
         schema = self.interface.get_source_schema()
         expected_description = "Path to directory containing OpenEphys legacy files."
         self.assertEqual(schema["properties"]["folder_path"]["description"], expected_description)
+
+    def test_openephyslegacy_date_parse(self):
+        extractor = self.interface.recording_extractor
+        neo_reader = extractor.neo_reader
+
+        blocks = neo_reader.raw_annotations.get("blocks", [])
+        segments = blocks[0].get("segments", [])
+        # Override date_created to test date is parsed
+        segments[0].update(date_created="29-Apr-2020 16554")
+        metadata = self.interface.get_metadata()
+        assert "session_start_time" in metadata["NWBFile"]
+        self.assertEqual(metadata["NWBFile"]["session_start_time"], datetime(2020, 4, 29, 16, 55, 4))
+

